src/gui/editor/Editor.ts

```diff
@@ -158,12 +158,43 @@ export class Editor implements ImageHandler, Component {
 			}
 		})
 
+		// Scroll to the cursor if it moves out of view
+		this.squire.addEventListener("cursor", () => {
+			const scrollableContainer = Editor.findParentByClass(domElement, "scroll")
+			if (scrollableContainer == null) return
+
+			const cursorPosition = this.getCursorPosition()
+
+			// Get whether the cursors edges have exceeded the bounds of the dialog
+			const offsetBottom = scrollableContainer.offsetTop + scrollableContainer.offsetHeight
+			if (cursorPosition.top <= scrollableContainer.offsetTop || cursorPosition.bottom >= offsetBottom) {
+				// Scroll to the cursor minus an arbitrary offset for aesthetic reasons
+				const offset = 10
+				scrollableContainer.scrollTo(cursorPosition.x - offset, cursorPosition.y)
+			}
+		})
+
 		this.domElement = domElement
 		// the _editor might have been disabled before the dom element was there
 		this.setEnabled(this.enabled)
 		this.initialized.resolve()
 	}
 
+	// Finds the nearest parent element of the element `element` and returns it
+	private static findParentByClass(element: HTMLElement, targetClass: string) {
```

Comment:
may I suggest [Element.closest()](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) ? Maybe you have already considered it, then it probably makes sense to document it

Reply:
Thank you, it's name totally slipped my mind. I will use `Element.closest()` instead.

Reply:
I didn't know it exists so I've also learned something :smile: